### PR TITLE
refactor!: remove deprecated GeneratedVaadinTimePicker

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

1. Removed deprecated `GeneratedVaadinTimePicker` class,
2. Updated `TimePicker` to implement `HasThemeVariant`.

## Type of change

- Breaking change

### CODE REVIEW

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java [hunk 1]
```diff
@@ -59,11 +66,18 @@
...
+@JsModule("@vaadin/time-picker/src/vaadin-time-picker.js")
 @JsModule("./vaadin-time-picker/timepickerConnector.js")
-public class TimePicker extends GeneratedVaadinTimePicker<TimePicker, LocalTime>
-        implements HasSize, HasValidation, HasEnabled, HasHelper, HasLabel,
-        HasTheme, HasClearButton, HasAllowedCharPattern,
-        HasValidator<LocalTime>, HasClientValidation, HasTooltip {
+public class TimePicker
+        extends AbstractSinglePropertyField<TimePicker, LocalTime>
+        implements Focusable<TimePicker>, HasAllowedCharPattern, HasClearButton,
+        HasClientValidation, HasEnabled, HasHelper, HasLabel, HasSize, HasStyle,
```
Comment (tomivirkki) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java:78]: `HasEnabled` is now redundant:

Screenshot 2022-12-13 at 13 27 11

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java [hunk 2]
```diff
@@ -23,23 +23,30 @@
 import java.util.stream.Collectors;
```
Comment (tomivirkki) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java:23]: Unused import

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java [hunk 3]
```diff
@@ -23,23 +23,30 @@
...
-import com.vaadin.flow.component.shared.HasClearButton;
-import com.vaadin.flow.component.shared.HasAllowedCharPattern;
-import com.vaadin.flow.component.shared.HasTooltip;
+import com.vaadin.flow.component.Focusable;
 import com.vaadin.flow.component.HasEnabled;
 import com.vaadin.flow.component.HasHelper;
 import com.vaadin.flow.component.HasLabel;
 import com.vaadin.flow.component.HasSize;
+import com.vaadin.flow.component.HasStyle;
 import com.vaadin.flow.component.HasTheme;
```
Comment (tomivirkki) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java:36]: Unused import

**Review by tomivirkki — APPROVED** [review]

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java (hunk 1 above)
Comment (web-padawan) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: Removed.

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java (hunk 2 above)
Comment (web-padawan) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: Removed.

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java (hunk 3 above)
Comment (web-padawan) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java]: Removed.

**Pull request merged**